src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AdditionalCondition14bHelper.kt

```diff
@@ -0,0 +1,191 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalConditionData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.Licence
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+private val dateTimeFormatter = DateTimeFormatter.ofPattern("EEEE dd MMMM yyyy")!!
+private const val conditionCode14B = "524f2fd6-ad53-47dd-8edc-2161d3dd2ed4"
+private const val ledDateChangeMsg = "licence end date"
+private const val crdOrArdChangeMsg = "release date"
+private const val ledAndCrdOrArdChangeMsg = "release date and licence end date"
+private const val END_Date = "endDate"
+
+fun updateAdditionalConditionWithAdditionConditionData(
+  sentenceChanges: SentenceChanges,
+  updatedLicenceEntity: Licence,
+  licenceEntity: Licence
+): AdditionalCondition? {
+  var endDate: String? = null
+  val additionalCondition14b = hasExistingLicenseContains14B(licenceEntity)
+  additionalCondition14b?.let {
+    if (hasCorrectStatusCodeForExisting14B(licenceEntity)) {
+      endDate = calculateEndDateForLedArdCrdChanges(sentenceChanges, updatedLicenceEntity, licenceEntity)
+    }
+    if (!endDate.isNullOrBlank()) {
```

Comment:
you could push `hasCorrectStatusCodeForExisting14B` into the `calculateEndDateForLedArdCrdChanges` function? 

`var` is a code smell and should generally be avoided unless there's a really exceptional reason.

Reply:
Yes we can have situations when date can be null when the business requirement has not been meet i.e the license may not be of correct status code  or When CRD has change but exiting LED <12 months and more stuff like that